config.go

```diff
@@ -492,8 +576,134 @@ func supportedSubsystems() []string {
+// function depending on the lookup configuration options. Lookup can be done
+// via golang's system resolver or via Tor. When Tor is used, only IPv4
+// addresses are returned.
+func lndLookup(host string) ([]string, error) {
+	return cfg.lookup(host)
+}
+
+// lndLookupSRV queries a DNS server with SRV requests depending on the
+// configuration options. SRV queries can be done via golang's system
+// resolver or through (but not by!) Tor.
+func lndLookupSRV(service, proto, name string) (string, []*net.SRV, error) {
+	return cfg.lookupSRV(service, proto, name)
+}
+
+// lndResolveTCP resolves TCP addresses into type *net.TCPAddr. Depending on
+// configuration options, this resolution can be done via golang's system
+// resolver or via Tor.
+func lndResolveTCP(network, address string) (*net.TCPAddr, error) {
+	return cfg.resolveTCP(network, address)
+}
+
+// proxyLookup uses Tor to resolve DNS via the SOCKS extension they provide for
+// resolution over the Tor network. Only IPV4 is supported. Unlike btcd's
+// TorLookupIP function, however, this function returns a ([]string, error)
+// tuple.
+func proxyLookup(host string, proxy string) ([]string, error) {
+	ip, err := connmgr.TorLookupIP(host, proxy)
+	if err != nil {
+		return nil, err
+	}
+
+	var addrs []string
+	// Only one IPv4 address is returned by the TorLookupIP function.
+	addrs = append(addrs, ip[0].String())
+	return addrs, nil
+}
+
+// proxyTCP uses Tor's proxy to resolve tcp addresses instead of the system resolver
+// that ResolveTCPAddr and related functions use. This resolver only queries DNS
+// servers in the case a hostname is passed in the address parameter. Only TCP
+// resolution is supported.
+func proxyTCP(address string, lookupFn func(string) ([]string, error)) (*net.TCPAddr, error) {
+	// Split host:port since the lookup function does not take a port.
+	host, port, err := net.SplitHostPort(address)
+	if err != nil {
+		return nil, err
+	}
+
+	// Look up the host's IP address via Tor.
+	ip, err := lookupFn(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert port to an int
+	p, err := strconv.Atoi(port)
+	if err != nil {
+		return nil, err
+	}
+
+	// Return a pointer to a net.TCPAddr struct exactly like ResolveTCPAddr.
+	return &net.TCPAddr{
+		IP:   net.ParseIP(ip[0]),
+		Port: p,
+	}, nil
+}
+
+// proxySRV uses Tor's proxy to route DNS SRV requests. Tor does not support
+// SRV queries. Therefore, we must route all SRV requests THROUGH the Tor proxy
+// and connect directly to a DNS server and query it. Note: the DNS server must
+// be a Lightning DNS server that follows the BOLT#10 specification.
+func proxySRV(dialer proxy.Dialer, service, proto, name string) (string, []*net.SRV, error) {
+
+	// _service._proto.name as described in RFC#2782.
+	host := "_" + service + "._" + proto + "." + name + "."
+
+	// Dial Eugene's lseed.
+	conn, err := dialer.Dial("tcp", "108.26.210.110:8053")
```

Comment:
Is this intended to be here for release? There must be a better solution than hardcoding a DNS seeder.

Reply:
My DNS seed is a slightly modified `lseed` node, but this shouldn't be here, it should instead query a specific Lightning DNS seeder with port 53 open to TCP connections - this should likely be configurable. Thanks for the catch!